phys/module_shallowcu_driver.F

```diff
@@ -508,6 +606,84 @@ SUBROUTINE shallowcu_driver(                                       &
          CALL wrf_error_fatal('Lacking arguments for SHCU_NSCV in shallow cumulus driver')
       ENDIF
 
+!
+! PSU-DENG
+!
+  CASE (DENGSHCUSCHEME)
+      CALL wrf_debug(100,'in PSU-DENG Shallow Cu CPS')
+
+
+    !  IF( bl_pbl_physics == 11 ) THEN
+    !     DO j=jts,jte
+    !     DO i=its,ite
+    !     DO k=kts,kte
+    !        tke_scr(I,K,J) = tkeb(I,K+1,J)
+    !        kth_scr(I,K,J) = kth3d(I,K+1,J)
+    !        bbls_scr(I,K,J) = bbls3d(I,K,J)
+    !     ENDDO
+    !     ENDDO
+    !     ENDDO
+    !  ELSE IF ( bl_pbl_physics == 2 ) THEN   ! MYJ
+     IF ( bl_pbl_physics == 2 ) THEN   ! MYJ
+         DO j=jts,jte
+         DO i=its,ite
+         DO k=kts,kte
+            tke_scr(I,K,J) = tke_pbl(I,K,J)
+            kth_scr(I,K,J) = exch_h(I,K,J)
+            bbls_scr(I,K,J) = el_pbl(I,K,J)
+         ENDDO
+         ENDDO
+         ENDDO
+      ELSE IF ( bl_pbl_physics == 5 ) THEN   ! MYNN
+         DO j=jts,jte
+         DO i=its,ite
+         DO k=kts,kte
+            tke_scr(I,K,J) = 0.5 * qke(I,K,J)
+            kth_scr(I,K,J) = exch_h(I,K,J)
+            bbls_scr(I,K,J) = el_pbl(I,K,J)
+         ENDDO
+         ENDDO
+         ENDDO
+      ELSE
+         WRITE(message,*) 'PSU DENG ShCu currently does not support PBL option: ', &
+                            bl_pbl_physics
+         CALL wrf_error_fatal ( message )
+         STOP
+      ENDIF
+      CALL deng_shcu_driver(                                             &
+            IDS=ids,IDE=ide,JDS=jds,JDE=jde,KDS=kds,KDE=kde              &
+           ,IMS=ims,IME=ime,JMS=jms,JME=jme,KMS=kms,KME=kme              &
+           ,ITS=its,ITE=ite,JTS=jts,JTE=jte,KTS=kts,KTE=kte              &
+           ,DT=dt ,KTAU=itimestep ,DX=dx, xtime=xtime, gmt=gmt           &
+           ,XLV=XLV, XLS=XLS                                             &
+           ,XLV0=XLV0 ,XLV1=XLV1 ,XLS0=XLS0 ,XLS1=XLS1                   &
+           ,CP=CP, R=R_d ,G=G                                            &
+           ,SVP1=SVP1 ,SVP2=SVP2 ,SVP3=SVP3 ,SVPT0=SVPT0                 &
+           ,ADAPT_STEP_FLAG=ADAPT_STEP_FLAG, DSIGMA=dnw                  &
+           ,XLONG=xlong, HT=ht, PBLH=pblh   &
```

Comment:
line things up, here and the next 10-15 lines there are a few that would be nice